### PR TITLE
Feature: Add `sortByProp`

### PR DESCRIPTION
**Pull request by nolleto**
Implemented `sortByProp` function.

Issue: #1241

### CODE REVIEW

**Comment by nolleto** [comment]
Hey @char0n , can you review my PR? 🫣🥺

**Comment by char0n** [comment]
@nolleto working on it now. Thanks for contributing!

**Review by char0n — CHANGES_REQUESTED** [review]
Hi @nolleto, please have a look at my code review comments.

I've reworked the unitests, added typings and also covered typings with tests.

File: types/index.d.ts [hunk 1]
```diff
@@ -1703,6 +1703,12 @@ export function padCharsStart(
 export function padStart(targetLength: number, value: string): string;
 export function padStart(targetLength: number): (value: string) => string;
 
+/**
+ * Sort a list of objects by a property.
+ */
+export function sortByProp<T>(prop: keyof T | number, list: T[]): T[];
+export function sortByProp<T>(prop: keyof T | number): (list: T[]) => T[];
```
Comment (nolleto) [inline types/index.d.ts:1710]: Hey @char0n , I improve a little the types with generics.

**Review by char0n — APPROVED** [review]
Did some additional changes to types. The rest looks fine. Thanks for contributing!

File: types/index.d.ts (hunk 1 above)
Comment (char0n) [reply types/index.d.ts]: I did some final changes as well ;] Thanks

**Pull request merged**